src/cargo/ops/cargo_install.rs

```diff
@@ -239,55 +241,101 @@ fn install_one(
         }
     };
 
-    config.shell().status("Installing", pkg)?;
+    if !opts.filter.is_specific() && !pkg.targets().iter().any(|t| t.is_bin()) {
```

Comment:
This is probably copy/pasted from somewhere originally, and I'll probably figure this out later on, but as a note I'm not sure why these two clauses are together. The latter "check for no binaries" makes sense but I'm not sure why it's coupled with an `is_specific` check. Maybe add a comment here to indicate?